libc/src/link/linux/dl_iterate_phdr.cpp

```diff
@@ -0,0 +1,57 @@
+//===-- Implementation of dl_iterate_phdr ---------------------------------===//
```

Comment:
I thought that this PR was just to make libunwind work (per PR description), hence `dladdr` being a stub, but we have a real implementation here for `dl_iterate_phdr`?

Reply:
It's because `dl_iterate_phdr` is simple enough to implement. `dladdr` will require more stuff like `dlopen`, `dlclose`, and `dlsym` to be implemented. Currently, there's *TODO*'s for @izaakschroeder to work on this.

Reply:
See the [`dlfcn.h` placeholder PR](https://github.com/llvm/llvm-project/pull/97501) and various related issues:

- https://github.com/llvm/llvm-project/issues/97917
- https://github.com/llvm/llvm-project/issues/97918
- https://github.com/llvm/llvm-project/issues/97919
- https://github.com/llvm/llvm-project/issues/97920

Reply:
Just found this #97928